jobrunner/job_executor.py

```diff
@@ -131,9 +131,9 @@ def from_dict(cls, metadata: dict):
         except (TypeError, ValueError):
             exit_code = None
         return cls(
-            outputs=metadata["outputs"],
-            unmatched_patterns=metadata["unmatched_patterns"],
-            unmatched_outputs=metadata["unmatched_outputs"],
+            outputs=metadata.get("outputs"),
+            unmatched_patterns=metadata.get("unmatched_patterns"),
+            unmatched_outputs=metadata.get("unmatched_outputs"),
```

Comment:
This tingles my spider-senses a bit in that it indicates that we have JobResults which both do and don't contain file information and that doesn't seem good. I wonder if it's because we're using the same class in two places where we really want two classes. I'll need to poke around a bit more to understand what's happening here.

Reply:
I think that we should get rid of JobResults and just use what we need from Task.agent_results directly.

It's only still there because some the existing APIs used it, and we wanted to do minimal changes.  I think it should go at some point soon.